### PR TITLE
Add citation info

### LINKED ISSUE #132: Update citation and CITATION.ctf
Since we have a main OGL paper now the citation section of README.md should reflect this. Also we should add a CITATION.ctf  file with this reference. An example of a citation.ctf can be found here. The link to the paper is available here:

https://link.springer.com/article/10.1007/s11012-024-01806-1

### PR DESCRIPTION
**Pull request by chihtaw**
1. Update citation info in README
2. Add a file for citation

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - family-names: "Olenik"
+    given-names: "Gregor"
+  - family-names: "Koch"
+    given-names: "Marcel"
+  - family-names: "Boutanios"
+    given-names: "Ziad"
+  - family-names: "Anzt"
+    given-names: "Hartwig" 
+  doi: "10.1007/s11012-024-01806-1"
+  journal: " Meccanica"
```
Comment (greole) [inline CITATION.cff]:
```suggestion
  journal: "Meccanica"
```

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+message: "If you use this software, please cite it as below."
+authors:
+  - family-names: "Olenik"
+    given-names: "Gregor"
+  - family-names: "Koch"
+    given-names: "Marcel"
+  - family-names: "Boutanios"
+    given-names: "Ziad"
+  - family-names: "Anzt"
+    given-names: "Hartwig"
```
Comment (greole) [inline CITATION.cff]:
```suggestion
- name "The OGL authors"
```

File: README.md [hunk 3]
```diff
@@ -143,6 +143,19 @@ title = {{Ginkgo: A Modern Linear Operator Algebra Framework for High Performanc
...
+author={Olenik, Gregor and Koch, Marcel and Boutanios, Ziad and Anzt, Hartwig},
+title={Towards a platform-portable linear algebra backend for OpenFOAM},
+journal={Meccanica},
+year={2024},
+month={Jun},
+day={12},
+issn={1572-9648},
+doi={10.1007/s11012-024-01806-1},
+url={https://doi.org/10.1007/s11012-024-01806-1}
+}
```
Comment (greole) [inline README.md:145]: Please move this above the Ginkgo paper

**Review by greole — CHANGES_REQUESTED** [review]

File: CITATION.cff (hunk 1 above)
Comment (chihtaw) [reply CITATION.cff]: Done

File: CITATION.cff (hunk 2 above)
Comment (chihtaw) [reply CITATION.cff]: Update as requested

File: README.md (hunk 3 above)
Comment (chihtaw) [reply README.md:145]: Done

**Review by greole — APPROVED** [review]

**Pull request merged**